core-c-bridge/src/callback_based_worker_client/interop.rs

```diff
@@ -0,0 +1,264 @@
+        callback_ud:    *mut libc::c_void,
+    ),
+
+    pub fail_nexus_task: unsafe extern "C" fn(
+        user_data:      *mut libc::c_void,
+        request:        ByteArrayRef,
+        callback:       WorkerClientCallbackTrampoline,
+        callback_ud:    *mut libc::c_void,
+    ),
+
+    pub get_workflow_execution_history: unsafe extern "C" fn(
+        user_data:      *mut libc::c_void,
+        request:        ByteArrayRef,
+        callback:       WorkerClientCallbackTrampoline,
+        callback_ud:    *mut libc::c_void,
+    ),
+
+    pub respond_legacy_query: unsafe extern "C" fn(
+        user_data:      *mut libc::c_void,
+        request:        ByteArrayRef,
+        callback:       WorkerClientCallbackTrampoline,
+        callback_ud:    *mut libc::c_void,
+    ),
+
+    pub describe_namespace: unsafe extern "C" fn(
+        user_data:      *mut libc::c_void,
+        request:        ByteArrayRef,
+        callback:       WorkerClientCallbackTrampoline,
+        callback_ud:    *mut libc::c_void,
+    ),
+
+    pub shutdown_worker: unsafe extern "C" fn(
+        user_data:           *mut libc::c_void,
+        request:             ByteArrayRef,
+        callback:            WorkerClientCallbackTrampoline,
+        callback_user_data:  *mut libc::c_void,
+    ),
+
+    pub is_mock: unsafe extern "C" fn(
+        user_data: *mut libc::c_void,
+    ) -> bool,
+
+    // The [temporal_sdk_core::WorkerClient] `workers()`, `capabilities()`, and `sdk_name_and_version()` functions are purely implemented in Rust, no interop implementation is offered.
+    // The [temporal_sdk_core::WorkerClient] `replace_client()` function is not supported by the [CallbackBasedWorkerClient].
+}
+
+unsafe impl Send for CallbackBasedWorkerClientRPCs {}
+unsafe impl Sync for CallbackBasedWorkerClientRPCs {}
+
+/// Create a new [Worker] with a custom callback-backed [temporal_sdk_core::WorkerClient] implementation.
+///
+/// # Safety
+/// - All pointer args (`client_rpcs`, `options`, `client_config`, `runtime`) must be non-null and valid.
+/// - `system_info_ref.data` must point to at least `system_info_ref.size` bytes if non-zero.
+/// - Caller retains ownership; these must outlive the call.
+///
+/// # Params
+/// - `client_rpcs`: pointer to [CallbackBasedWorkerClientRPCs] implementing Core SDK [temporal_sdk_core::WorkerClient] RPCs.
+/// - `options`:     pointer to [WorkerOptions].
+/// - `client_config`: pointer to [CallbackBasedWorkerClientConfig] (client name, version, identity).
+/// - `system_info_ref`: zero-copy [GetSystemInfoResponse] bytes or empty.
+/// - `runtime`:        pointer to [Runtime].
+///
+/// # Returns
+/// A [WorkerOrFail] struct containing:
+/// - `worker`: on success, a heap-allocated [Worker] pointer; null on error.
+/// - `fail`: on error, a C string (UTF-8) describing the failure; null on success.
+#[unsafe(no_mangle)]
+pub extern "C" fn temporal_core_new_callback_based_worker_client(
```

Comment:
```suggestion
pub extern "C" fn temporal_core_worker_callback_based_new(
```

or

```suggestion
pub extern "C" fn temporal_core_callback_based_worker_new(
```

Pedantic, but just clearer that you're creating a worker, not a client